release/releaser.go

```diff
@@ -154,6 +154,16 @@ func (r *Releaser) release(instanceID flux.InstanceID, job *jobs.Job, logStatus
 	applyErr := applyChanges(rc.Instance, updates, results)
 	timer.ObserveDuration()
 
+	// #492: If updating all services, but requested a specific image, only result in services that pertain to that image
+	if spec.ServiceSpec == flux.ServiceSpecAll && spec.ImageSpec != flux.ImageSpecNone {
+		for k, v := range results {
+			if v.Status == flux.ReleaseStatusSkipped || v.Status == flux.ReleaseStatusIgnored {
+				delete(results, k)
+			}
+		}
+
+	}
```

Comment:
This actually brought up another issue.
What happens when `--all` and `--latest`? I'm assuming you want to see everything then.